packages/loader/container-loader/src/container.ts

```diff
@@ -823,6 +823,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                             summary,
                             createNewResolvedUrl,
                             this.subLogger,
+                            !this.client.details.capabilities.interactive,
                         ),
                         "containerAttach",
                         this.mc.logger,
```

Comment:
Here, I think we should assert that it's never a summarizer. At least our current design does not allow start of summarizer instance before container is actually created in storage, and that's creation path.
Being more explicit helps readers, and enforcing invariants helps find bugs in the future.

Which brings to main point - that change is mostly noop, and actual changes are missing, for example 1061, where we call this.serviceFactory.createDocumentService.

I'd suggest you make these args not optional temporarily (on your box) and fix all the places in the code to explicitly pass arguments everywhere in client code base and in drivers. That way you will find all such cases (like mentioned above), and prep code base to make it non-optional in the future (I think it would be great to move to such world).